docs/proposals/scheduling/20230707-eviction-arbitrate-mechanism-in-descheduler.md

```diff
@@ -0,0 +1,256 @@
+2. Describe a simple central flow control mechanism to limit the number of Pod migrations over a period of time.
+3. Describe in detail the design details of the arbitrate mechanism.
+
+### Non-Goals/Future Work
+
+1. A new CRD or Controller.
+2. The generation process of PodMigrationJob.
+3. The specific execution process of PodMigrationJob.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+The descheduler evicts Pods with different priorities, and the importance of high priority Pods is also high. Therefore, The cost of migrating high priority Pods is greater than that of low priority Pods. So users expect low-priority Pods to be executed first to minimize migration costs.
+
+#### Story 2
+
+Multiple Pods of different workloads may be simultaneously evicted by the descheduler. We hope that the migrated Pods can be dispersed across different workloads (like deployments) as much as possible, in order to avoid a decrease in service availability caused by the migration process.
+
+#### Story 3
+
+Multiple Pods in different jobs are evicted. We should migrate all Pods from one job at a time and process different jobs in sequence to avoid a thundering herd effect.
+
+### Pod Migration Job CRD Field
+
+In order to support the above user stories, we add some fileds in Custom Resource Definition(CRD) `PodMigrationJob`.
+
+We can use `PriorityClassName` and `Priority` fields to customize the priority of migration tasks, if we want a migration priority that is different from the pod priority.
+
+#### Migration Job Spec
+
+```go
+type PodMigrationJobSpec struct {
+    // PriorityClassName defines the priority of the PodMigrationJob which should be 
+    // the name of some PriorityClass object.
+    // +optional
+    PriorityClassName string `json:"priorityClassName,omitempty"`
+    
+    // Priority define the priority of the PodMigrationJob.
+    // If this field is set, PriorityClassName will be discarded.
+    // +optional
+    Priority          *int32 `json:"priority,omitempty"`
+}
+```
+
+- `PriorityClassName` indicates the priority of the PodMigrationJob which should be the name of some PriorityClass object, and the Priority value will be set as the value of the PriorityClass.
+- `Priority` indicate the priority of the PodMigrationJob. The higher the priority, the greater the possibility of arbitration. If this field is set, PriorityClassName will be discarded.
+
+### Implementation Details/Notes/Constraints
+
+#### PodMigrationJob Controller
+
+The **PodMigrationJobController** will evaluate all PodMigrationJobs and select a batch of PodMigrationJob and execute them. This selection process is called the **arbitration mechanism**. The reason why the arbitration mechanism is introduced is mainly to control the stability risk and control the cost of migrating Pods. The arbitration mechanism includes three stages: `Sort`, `GroupFilter` and `Select`.
```

Comment:
> The reason why the arbitration mechanism is introduced is mainly to control the stability risk and control the cost of migrating Pods.

I think we have described the motivation clearly enough that we can be concise and not redundant.

Reply:
You are right, this sentence should be deleted.